Unit Test Flake: TestBlobDescriptorServiceIsApplied

Seen [here](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check_future/915/consoleFull#-178241454456bf4006e4b05b79524e5923):

```
00:41:56.633 Of 5059 tests executed in 2230.745s, 5039 succeeded, 1 failed, and 19 were skipped.
00:41:56.633 In suite "github.com/openshift/origin/pkg/dockerregistry/server", test case "TestBlobDescriptorServiceIsApplied" failed:
00:41:56.633 === RUN   TestBlobDescriptorServiceIsApplied
00:41:56.633 time="2017-03-07T17:31:29Z" level=info msg="OpenShift middleware for storage driver initializing" 
00:41:56.633 time="2017-03-07T17:31:29Z" level=warning msg="No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable." go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="redis not configured" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="Starting upload purge in 32m0s" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="using inmemory blob descriptor cache" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="OpenShift registry middleware initializing" 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="quota enforcement disabled" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.634 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=a022a63d-1140-4506-a23d-33e2fc41d704 http.request.method=POST http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/uploads/" http.request.useragent="Go-http-client/1.1" http.response.duration=11.831003ms http.response.status=202 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.640 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.contenttype="application/octet-stream" http.request.host="127.0.0.1:41357" http.request.id=36000dcc-bb78-4c49-84ed-abc479df611c http.request.method=PATCH http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/uploads/3c49dc8a-d77b-4110-a45b-356bf7047b2d?_state=wzSARGSU7LVxzT6o0pmelczyNS0LO53WnwaJuF-4Rb57Ik5hbWUiOiJ1c2VyL2FwcCIsIlVVSUQiOiIzYzQ5ZGM4YS1kNzdiLTQxMTAtYTQ1Yi0zNTZiZjcwNDdiMmQiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMTctMDMtMDdUMTc6MzE6MjkuNDQ0OTI5MzE5WiJ9" http.request.useragent="Go-http-client/1.1" http.response.duration=13.547802ms http.response.status=202 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.641 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=97414fdc-138f-488c-a008-239f3d67f515 http.request.method=PUT http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/uploads/3c49dc8a-d77b-4110-a45b-356bf7047b2d?_state=zKbajMtaUuv0TdFT01p5my40Sd84kZva3xt_4NdCtCB7Ik5hbWUiOiJ1c2VyL2FwcCIsIlVVSUQiOiIzYzQ5ZGM4YS1kNzdiLTQxMTAtYTQ1Yi0zNTZiZjcwNDdiMmQiLCJPZmZzZXQiOjQwOTYsIlN0YXJ0ZWRBdCI6IjIwMTctMDMtMDdUMTc6MzE6MjlaIn0%3D&digest=sha256%3A7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.duration=14.758583ms http.response.status=201 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.642 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=777ae1ab-80c1-4d76-8e85-d85a40c4a990 http.request.method=HEAD http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/octet-stream" http.response.duration=13.108979ms http.response.status=200 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.642 time="2017-03-07T17:31:29Z" level=error msg="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=9e252bcb-387f-488d-b126-d326520743df http.request.method=GET http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.643 time="2017-03-07T17:31:29Z" level=error msg="Failed to find blob \"sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b\": &errors.errorString{s:\"failed to retrieve repository from context\"}" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=9e252bcb-387f-488d-b126-d326520743df http.request.method=GET http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.651 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code=unknown err.detail="failed to retrieve repository from context" err.message="unknown error" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=9e252bcb-387f-488d-b126-d326520743df http.request.method=GET http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=36.302323ms http.response.status=500 http.response.written=70 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.653 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=8b4226f6-f8d7-4eb8-a713-f8672054cc15 http.request.method=GET http.request.remoteaddr="127.0.0.1:40366" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/octet-stream" http.response.duration=29.528363ms http.response.status=200 http.response.written=4096 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.653 time="2017-03-07T17:31:29Z" level=error msg="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=c49b2352-e230-42f5-9cb4-20e1f2da8b29 http.request.method=HEAD http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.654 time="2017-03-07T17:31:29Z" level=error msg="Failed to find blob \"sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b\": &errors.errorString{s:\"failed to retrieve repository from context\"}" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=c49b2352-e230-42f5-9cb4-20e1f2da8b29 http.request.method=HEAD http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.654 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code=unknown err.detail="failed to retrieve repository from context" err.message="unknown error" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=c49b2352-e230-42f5-9cb4-20e1f2da8b29 http.request.method=HEAD http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=16.844408ms http.response.status=500 http.response.written=70 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.655 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=108f96a5-79a5-48ad-bd86-d46b1f341fe3 http.request.method=HEAD http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/octet-stream" http.response.duration=10.352401ms http.response.status=200 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.656 time="2017-03-07T17:31:29Z" level=error msg="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=5aabb4cf-d21c-40a3-8190-626d0b21f624 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.665 time="2017-03-07T17:31:29Z" level=error msg="Unknown error deleting blob: failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=5aabb4cf-d21c-40a3-8190-626d0b21f624 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.665 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code=unknown err.message="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=5aabb4cf-d21c-40a3-8190-626d0b21f624 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=9.759269ms http.response.status=500 http.response.written=70 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.digest="sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" vars.name="user/app" 
00:41:56.666 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=1bfe455e-a96a-4d41-ba65-84f36112d635 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/blobs/sha256:7a22b22a70ddc8deaaa503169e8564386d5be928d427f34395361b75e3fd203b" http.request.useragent="Go-http-client/1.1" http.response.duration=9.223895ms http.response.status=202 http.response.written=0 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.666 time="2017-03-07T17:31:29Z" level=info msg="(*repository).getImage: got image sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.667 time="2017-03-07T17:31:29Z" level=error msg="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=10743007-f18e-4c0b-9998-e507b3b315eb http.request.method=GET http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/manifests/latest" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference=latest 
00:41:56.667 time="2017-03-07T17:31:29Z" level=error msg="unable to get manifest from storage: failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=10743007-f18e-4c0b-9998-e507b3b315eb http.request.method=GET http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/manifests/latest" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference=latest 
00:41:56.667 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code="manifest unknown" err.detail="failed to retrieve repository from context" err.message="manifest unknown" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=10743007-f18e-4c0b-9998-e507b3b315eb http.request.method=GET http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/manifests/latest" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=9.592699ms http.response.status=404 http.response.written=82 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference=latest 
00:41:56.668 time="2017-03-07T17:31:29Z" level=info msg="(*repository).getImage: got image sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" go.version=go1.7.4 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.668 time="2017-03-07T17:31:29Z" level=info msg="response completed" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=0dc6eabb-e43b-4476-a3e3-af08026c76a0 http.request.method=GET http.request.remoteaddr="127.0.0.1:40368" http.request.uri="/v2/user/app/manifests/latest" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/vnd.docker.distribution.manifest.v1+prettyjws" http.response.duration=41.954176ms http.response.status=200 http.response.written=1130 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a 
00:41:56.669 time="2017-03-07T17:31:29Z" level=error msg="failed to retrieve repository from context" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=06d79268-2a78-4361-a736-c978531cd213 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40370" http.request.uri="/v2/user/app/manifests/sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" http.request.useragent="Go-http-client/1.1" instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference="sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" 
00:41:56.670 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code=unknown err.message="unknown error" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=06d79268-2a78-4361-a736-c978531cd213 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40370" http.request.uri="/v2/user/app/manifests/sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=10.477901ms http.response.status=500 http.response.written=58 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference="sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" 
00:41:56.670 time="2017-03-07T17:31:29Z" level=error msg="response completed with error" err.code="manifest unknown" err.message="manifest unknown" go.version=go1.7.4 http.request.host="127.0.0.1:41357" http.request.id=3eeb7106-5ee8-43d1-8157-d820359e90a1 http.request.method=DELETE http.request.remoteaddr="127.0.0.1:40370" http.request.uri="/v2/user/app/manifests/sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" http.request.useragent="Go-http-client/1.1" http.response.contenttype="application/json; charset=utf-8" http.response.duration=17.485246ms http.response.status=404 http.response.written=70 instance.id=2dd18ade-78f9-4778-b68e-17d0e59b7c1a vars.name="user/app" vars.reference="sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b" 
00:41:56.671 --- FAIL: TestBlobDescriptorServiceIsApplied (5.39s)
00:41:56.671 	blobdescriptorservice_test.go:486: instantiating fake access controller
00:41:56.671 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.671 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:push
00:41:56.671 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.671 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:push
00:41:56.671 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.672 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:push
00:41:56.672 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.672 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.672 	blobdescriptorservice_test.go:465: unsetting repository from the context
00:41:56.672 	blobdescriptorservice_test.go:158: [get blob with repository unset] received errors: errcode.Errors{errcode.Error{Code:1000, Message:"unknown error", Detail:map[string]interface {}{}}}
00:41:56.692 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.692 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.692 	blobdescriptorservice_test.go:465: unsetting repository from the context
00:41:56.692 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.692 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:*
00:41:56.692 	blobdescriptorservice_test.go:465: unsetting repository from the context
00:41:56.692 	blobdescriptorservice_test.go:158: [delete blob with repository unset] received errors: errcode.Errors{errcode.Error{Code:1000, Message:"unknown error", Detail:map[string]interface {}{}}}
00:41:56.693 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:*
00:41:56.693 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.693 	util.go:82: imagestream get handler: returning image stream user/app
00:41:56.693 	util.go:228: images get handler: returning image sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b
00:41:56.693 	blobdescriptorservice_test.go:465: unsetting repository from the context
00:41:56.693 	blobdescriptorservice_test.go:158: [get manifest with repository unset] received errors: errcode.Errors{errcode.Error{Code:1012, Message:"manifest unknown", Detail:map[string]interface {}{}}}
00:41:56.693 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:pull
00:41:56.693 	util.go:82: imagestream get handler: returning image stream user/app
00:41:56.693 	util.go:228: images get handler: returning image sha256:bff08d8da5a030fc39fdfcddd533fac4bc76d877d1a33bc336f8115e18d9081b
00:41:56.693 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:*
00:41:56.693 	blobdescriptorservice_test.go:465: unsetting repository from the context
00:41:56.693 	blobdescriptorservice_test.go:158: [delete manifest with repository unset] received errors: errcode.Errors{1000}
00:41:56.694 	blobdescriptorservice_test.go:499: fake authorizer: authorizing access to repository:user/app:*
00:41:56.694 	blobdescriptorservice_test.go:144: [delete manifest] unexpected status code: 404 != 202
00:41:56.694 	blobdescriptorservice_test.go:158: [delete manifest] received errors: errcode.Errors{1012}
00:41:56.694 	blobdescriptorservice_test.go:165: [delete manifest] failed to get stats: timeout while waiting on expected stats
```

--- a/pkg/dockerregistry/server/blobdescriptorservice_test.go
+++ b/pkg/dockerregistry/server/blobdescriptorservice_test.go
@@ -256,6 +256,33 @@ func TestBlobDescriptorServiceIsApplied(t *testing.T) {
 		},
 
 		{
+			name:     "delete manifest with repository unset",
+			method:   http.MethodDelete,
+			endpoint: v2.RouteNameManifest,
+			vars: []string{
+				"name", "user/app",
+				"reference", testImage.Name,
+			},
+			unsetRepository: true,
+			expectedStatus:  http.StatusInternalServerError,
+			// we don't allow to delete manifests from etcd; in this case, we attempt to delete layer link
+			expectedMethodInvocations: map[string]int{"Stat": 1},
+		},
+
+		{
+			name:     "delete manifest",
+			method:   http.MethodDelete,
+			endpoint: v2.RouteNameManifest,
+			vars: []string{
+				"name", "user/app",
+				"reference", testImage.Name,
+			},
+			expectedStatus: http.StatusNotFound,
+			// we don't allow to delete manifests from etcd; in this case, we attempt to delete layer link
+			expectedMethodInvocations: map[string]int{"Stat": 1},
+		},
+
+		{
 			name:     "get manifest with repository unset",
 			method:   http.MethodGet,
 			endpoint: v2.RouteNameManifest,
@@ -281,36 +308,6 @@ func TestBlobDescriptorServiceIsApplied(t *testing.T) {
 			expectedStatus: http.StatusOK,
 			// manifest is retrieved from etcd
 			expectedMethodInvocations: map[string]int{"Stat": 1},
-		},
-
-		{
-			name:     "delete manifest with repository unset",
-			method:   http.MethodDelete,
-			endpoint: v2.RouteNameManifest,
-			vars: []string{
-				"name", "user/app",
-				"reference", testImage.Name,
-			},
-			unsetRepository: true,
-			expectedStatus:  http.StatusInternalServerError,
-			// we don't allow to delete manifests from etcd; in this case, we attempt to delete layer link
-			expectedMethodInvocations: map[string]int{"Stat": 1},
-		},
-
-		{
-			name:     "delete manifest",
-			method:   http.MethodDelete,
-			endpoint: v2.RouteNameManifest,
-			vars: []string{
-				"name", "user/app",
-				"reference", testImage.Name,
-			},
-			expectedStatus: http.StatusAccepted,
-			// we don't allow to delete manifests from etcd; in this case, we attempt to delete layer link
-			expectedMethodInvocations: map[string]int{
-				"Stat":  1,
-				"Clear": 1,
-			},
 		},
 	} {
 		doTest(tc)

--- a/pkg/dockerregistry/server/blobdescriptorservice_test.go
+++ b/pkg/dockerregistry/server/blobdescriptorservice_test.go
@@ -77,6 +77,11 @@ func TestBlobDescriptorServiceIsApplied(t *testing.T) {
 			"delete": configuration.Parameters{
 				"enabled": true,
 			},
+			"maintenance": configuration.Parameters{
+				"uploadpurging": map[interface{}]interface{}{
+					"enabled": false,
+				},
+			},
 		},
 		Middleware: map[string][]configuration.Middleware{
 			"registry":   {{Name: "openshift"}},

--- a/pkg/dockerregistry/server/pullthroughblobstore_test.go
+++ b/pkg/dockerregistry/server/pullthroughblobstore_test.go
@@ -14,8 +14,6 @@ import (
 	"testing"
 	"time"
 
-	log "github.com/Sirupsen/logrus"
-
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/context"
 	"github.com/docker/distribution/digest"
@@ -31,7 +29,6 @@ import (
 func TestPullthroughServeBlob(t *testing.T) {
 	namespace, name := "user", "app"
 	repoName := fmt.Sprintf("%s/%s", namespace, name)
-	log.SetLevel(log.DebugLevel)
 	installFakeAccessController(t)
 	setPassthroughBlobDescriptorServiceFactory()
 
@@ -209,7 +206,6 @@ func TestPullthroughServeBlob(t *testing.T) {
 func TestPullthroughServeNotSeekableBlob(t *testing.T) {
 	namespace, name := "user", "app"
 	repoName := fmt.Sprintf("%s/%s", namespace, name)
-	log.SetLevel(log.DebugLevel)
 	installFakeAccessController(t)
 	setPassthroughBlobDescriptorServiceFactory()
 
@@ -367,7 +363,6 @@ func TestPullthroughServeBlobInsecure(t *testing.T) {
 	repo1Name := fmt.Sprintf("%s/%s", namespace, repo1)
 	repo2Name := fmt.Sprintf("%s/%s", namespace, repo2)
 
-	log.SetLevel(log.DebugLevel)
 	installFakeAccessController(t)
 	setPassthroughBlobDescriptorServiceFactory()
 

--- a/pkg/dockerregistry/server/pullthroughmanifestservice_test.go
+++ b/pkg/dockerregistry/server/pullthroughmanifestservice_test.go
@@ -8,8 +8,6 @@ import (
 	"strings"
 	"testing"
 
-	log "github.com/Sirupsen/logrus"
-
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/configuration"
 	"github.com/docker/distribution/context"
@@ -39,6 +37,11 @@ func createTestRegistryServer(t *testing.T, ctx context.Context) *httptest.Serve
 			"delete": configuration.Parameters{
 				"enabled": true,
 			},
+			"maintenance": configuration.Parameters{
+				"uploadpurging": map[interface{}]interface{}{
+					"enabled": false,
+				},
+			},
 		},
 	})
 
@@ -59,7 +62,6 @@ func TestPullthroughManifests(t *testing.T) {
 	repoName := fmt.Sprintf("%s/%s", namespace, repo)
 	tag := "latest"
 
-	log.SetLevel(log.DebugLevel)
 	client := &testclient.Fake{}
 
 	installFakeAccessController(t)
@@ -179,7 +181,6 @@ func TestPullthroughManifestInsecure(t *testing.T) {
 	repo := "zapp"
 	repoName := fmt.Sprintf("%s/%s", namespace, repo)
 
-	log.SetLevel(log.DebugLevel)
 	installFakeAccessController(t)
 	setPassthroughBlobDescriptorServiceFactory()
 

--- a/pkg/dockerregistry/server/repositorymiddleware_test.go
+++ b/pkg/dockerregistry/server/repositorymiddleware_test.go
@@ -40,6 +40,10 @@ const (
 	testImageLayerCount = 2
 )
 
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
 func TestRepositoryBlobStat(t *testing.T) {
 	quotaEnforcing = &quotaEnforcingConfig{}
 
@@ -326,7 +330,6 @@ func TestRepositoryBlobStat(t *testing.T) {
 }
 
 func TestRepositoryBlobStatCacheEviction(t *testing.T) {
-	log.SetLevel(log.DebugLevel)
 	const blobRepoCacheTTL = time.Millisecond * 500
 
 	quotaEnforcing = &quotaEnforcingConfig{}

--- a/pkg/dockerregistry/server/signaturedispatcher_test.go
+++ b/pkg/dockerregistry/server/signaturedispatcher_test.go
@@ -75,6 +75,11 @@ func TestSignatureGet(t *testing.T) {
 			"delete": configuration.Parameters{
 				"enabled": true,
 			},
+			"maintenance": configuration.Parameters{
+				"uploadpurging": map[interface{}]interface{}{
+					"enabled": false,
+				},
+			},
 		},
 		Middleware: map[string][]configuration.Middleware{
 			"registry":   {{Name: "openshift"}},
@@ -172,6 +177,11 @@ func TestSignaturePut(t *testing.T) {
 			},
 			"delete": configuration.Parameters{
 				"enabled": true,
+			},
+			"maintenance": configuration.Parameters{
+				"uploadpurging": map[interface{}]interface{}{
+					"enabled": false,
+				},
 			},
 		},
 		Middleware: map[string][]configuration.Middleware{

